plugin/parse.go

```
@@ -0,0 +1,152 @@
+	json := jsoniter.ConfigCompatibleWithStandardLibrary
+	if directivesString, ok := v.AsMap()["directives"].(string); ok {
+		var wafDirectives WafDirectives
+		err := json.UnmarshalFromString(directivesString, &wafDirectives)
+		if err != nil {
+			return nil, err
+		}
+		if len(wafDirectives) == 0 {
+			return nil, errors.New("directives is empty")
+		}
+		config.directives = wafDirectives
+	} else {
+		return nil, errors.New("directives is not exist")
+	}
+	if defaultDirectiveString, ok := v.AsMap()["default_directive"].(string); ok {
+		_, ok := config.directives[defaultDirectiveString]
+		if !ok {
+			return nil, errors.New("default_directive is not exist")
+		}
+		config.defaultDirective = defaultDirectiveString
+	} else {
+		return nil, errors.New("default_directive is not exist")
+	}
+
+	//最好只是在ftw测试的时候使用
+	if auditLogPath, ok := v.AsMap()["audit_log_path"].(string); ok {
+		config.auditLogPath = auditLogPath
+	}
+
+	if hostDirectiveMapString, ok := v.AsMap()["host_directive_map"].(string); ok {
+		hostDirectiveMap := make(HostDirectiveMap)
+		err := json.UnmarshalFromString(hostDirectiveMapString, &hostDirectiveMap)
+		if err != nil {
+			return nil, err
+		}
+		for host, rule := range hostDirectiveMap {
+			_, ok := config.directives[rule]
+			if !ok {
+				return nil, errors.New(fmt.Sprintf("The rule corresponding to %s does not exist", host))
+			}
+		}
+		config.hostDirectiveMap = hostDirectiveMap
+		wafMaps := make(wafMaps)
+		for wafName, wafRules := range config.directives {
+			wafConfig := coraza.NewWAFConfig()
+			if len(config.auditLogPath) != 0 {
+				wafConfig = wafConfig.WithErrorCallback(logError(config.auditLogPath))
+			}
+			wafConfig = wafConfig.WithDirectives(strings.Join(wafRules.SimpleDirectives, "\n"))
+			for _, val := range wafRules.DirectivesFiles {
+				wafConfig = wafConfig.WithDirectivesFromFile(val)
+			}
+			waf, err := coraza.NewWAF(wafConfig)
+			if err != nil {
+				panic(fmt.Sprintf("%s mapping waf init error", wafName))
```

Better to return an error here